### PR TITLE
optimize imports

### PR DESCRIPTION
**Pull request by jpo2433**

### CODE REVIEW

**Comment by lusbuab** [comment]
👋

File: app/src/main/java/org/mobilitychoices/entities/End.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.mobilitychoices.entities;
+
+public class End {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/entities/End.java]: WTF? same as `Start` 😉

File: app/src/main/java/org/mobilitychoices/GeocoderTask.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            System.out.println("Geocoder loop: " + t.getId());
+            Geocoder geocoder;
+            List<Address> addresses1;
+            List<Address> addresses2;
+            geocoder = new Geocoder(context);
+            try {
+                addresses1 = geocoder.getFromLocation(t.getStart().getLatitude(),
+                        t.getStart().getLongitude(), 1);
+                String address = addresses1.get(0).getThoroughfare();
+                String city = addresses1.get(0).getLocality();
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/GeocoderTask.java]:
```java
List<Address> startAddress = geocoder.getFromLocation(t.getStart().getLatitude(), t.getStart.getLongitude(), 1);
String street = startAddress.get(0).getThoroughfare();
String city = startAddress.get(0).getLocality();
```

same for `endAddress`

File: app/src/main/java/org/mobilitychoices/GeocoderTask.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    protected ArrayList<Track> doInBackground(ArrayList<Track>... arrayLists) {
+        ArrayList<Track> tracks = arrayLists[0];
+
+        for (Track t :
+                tracks) {
+            System.out.println("Geocoder loop: " + t.getId());
+            Geocoder geocoder;
+            List<Address> addresses1;
+            List<Address> addresses2;
+            geocoder = new Geocoder(context);
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/GeocoderTask.java]:
```java
Geocoder geocoder = new Geocoder(context);
```

File: app/src/main/java/org/mobilitychoices/GeocoderTask.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                String address = addresses1.get(0).getThoroughfare();
+                String city = addresses1.get(0).getLocality();
+                System.out.println(addresses1.get(0).toString());
+
+                t.getStart().setAddress(address + ", " + city);
+
+                addresses2 = geocoder.getFromLocation(t.getEnd().getLatitude(),
+                        t.getEnd().getLongitude(), 1);
+                String address2 = addresses2.get(0).getThoroughfare();
+                String city2 = addresses2.get(0).getLocality();
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/GeocoderTask.java]: see above

File: app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java [hunk 5]
```diff
@@ -1,32 +1,114 @@
...
+
+    private void requestTracks(String token) {
+        new GetAllTracksTask(new GetAllTracksTask.IGetAllTracksCallback() {
+            @Override
+            public void done(Response<Object> response) {
+                if (response != null) {
+                    if (response.getCode() == 200) {
+                        showTracks((TrackList) response.getData());
+                    } else {
+                        if (response.getCode() == 400) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java]: → 
```java
} else if (...) {
```

File: app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java [hunk 6]
```diff
@@ -1,32 +1,114 @@
...
+        }, token).execute();
+    }
+
+    private void showTracks(TrackList data) {
+
+        ArrayList<JSONObject> Jtracks = data.getRoutes();
+        ArrayList<Track> tracks = new ArrayList<>();
+
+        for (JSONObject t :
+                Jtracks) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java]: format

```java
for (JSONObject t : Jtracks) {
```

File: app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java [hunk 7]
```diff
@@ -1,32 +1,114 @@
...
+                        }
+                    }
+                }
+            }
+        }, token).execute();
+    }
+
+    private void showTracks(TrackList data) {
+
+        ArrayList<JSONObject> Jtracks = data.getRoutes();
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java]: → `jTracks`

File: app/src/main/res/layout/content_all_routes_activty.xml [hunk 8]
```diff
@@ -13,4 +13,11 @@
     tools:context="org.mobilitychoices.activities.AllRoutesActivty"
     tools:showIn="@layout/activity_all_routes_activty">
 
+    <ListView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="40dp"
```
Comment (lusbuab) [inline app/src/main/res/layout/content_all_routes_activty.xml]: marginTop

File: app/src/main/java/org/mobilitychoices/remote/GetTrackTask.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    public GetTrackTask(IGetTrackCallback callback, String token, String id) {
+        iGetTracksCallback = callback;
+        this.token = token;
+        this.id = id;
+    }
+
+    @Override
+    protected Response<Object> doInBackground(Void... voids) {
+        String urlString = "/tracks/"+ id;
+        System.out.println("GetTrackTask in Background...");
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/remote/GetTrackTask.java]: remove sysout

File: app/src/main/java/org/mobilitychoices/entities/Track.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+            JSONObject locationJSON = startJSON.getJSONObject("location");
+            start = new Point(locationJSON.getDouble("lat"), locationJSON.getDouble("lon"));
+            JSONObject endJSON = jsonObject.getJSONObject("end");
+            JSONObject locationEndJSON = endJSON.getJSONObject("location");
+            end = new Point(locationEndJSON.getDouble("lat"), locationEndJSON.getDouble("lon"));
+
+            JSONObject strings  = jsonObject.getJSONObject("strings");
+            duration = strings.getString("duration");
+            created = strings.getString("created");
+
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/entities/Track.java]: remove empty line

File: app/src/main/java/org/mobilitychoices/entities/Track.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Override
+    public void fromJSON(JSONObject jsonObject) {
+        try {
+            id = jsonObject.getString("id");
+            JSONObject startJSON = jsonObject.getJSONObject("start");
+            JSONObject locationJSON = startJSON.getJSONObject("location");
+            start = new Point(locationJSON.getDouble("lat"), locationJSON.getDouble("lon"));
+            JSONObject endJSON = jsonObject.getJSONObject("end");
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/entities/Track.java:45]: insert empty line between `start` and `end`

File: app/src/main/java/org/mobilitychoices/activities/RegisterActivity.java [hunk 12]
```diff
@@ -72,12 +72,12 @@ public void done(Response<Object> response) {
                             } else {
                                 if (response.getCode() == 400) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/RegisterActivity.java]: else-if

File: app/src/main/java/org/mobilitychoices/activities/MapsActivity.java [hunk 13]
```diff
@@ -75,20 +89,22 @@ public void onMapReady(GoogleMap googleMap) {
...
+        if (locations != null && locations.size() >= 1) {
             LatLng latLng = new LatLng(locations.get(0).getLatitude(), locations.get(0).getLongitude());
 
             CameraPosition cameraPosition = new CameraPosition.Builder().target(latLng).zoom(10).build();
             googleMap.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition));
 
             PolylineOptions po = new PolylineOptions().geodesic(true);
-            for (LatLng l:
+            for (LatLng l :
                     latLngs) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/MapsActivity.java]:
```java
for (LatLng l : latLngs) {
```

File: app/src/main/java/org/mobilitychoices/GeocoderTask.java [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        iGeocoderCallback = callback;
+        context = c;
+    }
+
+    @Override
+    protected ArrayList<Track> doInBackground(ArrayList<Track>... arrayLists) {
+        ArrayList<Track> tracks = arrayLists[0];
+
+        for (Track t :
+                tracks) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/GeocoderTask.java]: format

File: app/src/main/java/org/mobilitychoices/GeocoderTask.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            System.out.println("Geocoder loop: " + t.getId());
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/GeocoderTask.java]: sysout

File: app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java [hunk 16]
```diff
@@ -1,32 +1,111 @@
...
+        String token = sharedPref.getString("token", null);
+
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Track track = adapter.getItem(position);
+                new GetTrackTask(new GetTrackTask.IGetTrackCallback() {
+                    @Override
+                    public void done(Response<Object> result) {
+                        System.out.println(((FullTrack) result.getData()).toJSON());
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java]: sysout

File: app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java [hunk 17]
```diff
@@ -1,32 +1,111 @@
...
+            track.fromJSON(t);
+            track.getStart().setAddress(track.getStart().getLatitude() + "&" + track.getStart().getLongitude());
+            track.getEnd().setAddress(track.getEnd().getLatitude() + "&" + track.getEnd().getLongitude());
+            tracks.add(track);
+        }
+
+        new GeocoderTask(new GeocoderTask.IGeocoderCallback() {
+            @Override
+            public void done(ArrayList<Track> response) {
+                System.out.println(response.get(0).toJSON());
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/AllRoutesActivty.java]: sysout

File: app/src/main/java/org/mobilitychoices/activities/DirectionsActivity.java [hunk 18]
```diff
@@ -39,16 +43,38 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
...
+            Step current = steps.get(i);
+            Location location = new Location(current.getStart().getLatitude(), current.getStart().getLongitude(), 0, 0, 0);
+            locations.add(location);
+            if (current.getSubsteps() != null && current.getSubsteps().size() > 0) {
+                locations = createLocationsList(locations, current.getSubsteps());
+            }
+        }
+        Point end = steps.get(steps.size() - 1).getEnd();
+        Location endPoint = new Location(end.getLatitude(), end.getLongitude(), 0, 0, 0);
+        locations.add(endPoint);
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/DirectionsActivity.java]:
```java
locations.add(new Location(end.getLatitude(), end.getLongitude(), 0, 0, 0));
```

File: app/src/main/java/org/mobilitychoices/activities/DirectionsActivity.java [hunk 19]
```diff
@@ -66,7 +92,7 @@ private void requestRoutes() {
             public void done(Response<Object> response) {
                 if (response != null) {
                     if (response.getCode() == 200) {
-                        showAlternativeRoutes((Routes) response.getData());
+                        showAlternativeRoutes((RoutesList) response.getData());
                     } else {
                         if (response.getCode() == 400) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/DirectionsActivity.java]: else-if

File: app/src/main/java/org/mobilitychoices/activities/LoginActivity.java [hunk 20]
```diff
@@ -179,7 +179,7 @@ public void done(Response<Object> response) {
                     LoginActivity.this.finish();
                 } else {
                     Log.i("Login", "Login failed " + response.getCode());
-                    if(response.getCode() == 400){
+                    if (response.getCode() == 400) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/activities/LoginActivity.java]: else-if

File: app/src/main/java/org/mobilitychoices/entities/Entity.java [hunk 21]
```diff
@@ -3,6 +3,10 @@
 import org.json.JSONObject;
 
 public class Entity {
```
Comment (optionset) [inline app/src/main/java/org/mobilitychoices/entities/Entity.java]: abstract?

File: app/src/main/java/org/mobilitychoices/entities/FullTrack.java [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public JSONObject toJSON() {
+        JSONObject jsonObject = new JSONObject();
+
+        try {
+            jsonObject.put("owner", owner);
+            jsonObject.put("creted", created);
+            JSONArray locationsJson = new JSONArray();
+            for (Location l :
+                    locations) {
```
Comment (lusbuab) [inline app/src/main/java/org/mobilitychoices/entities/FullTrack.java]: for ...

File: app/src/main/java/org/mobilitychoices/entities/ListGenerator.java [hunk 23]
```diff
@@ -6,22 +6,24 @@
...
-    ArrayList<JSONObject> routes = new ArrayList<>();
+    private ArrayList<JSONObject> routes = new ArrayList<>();
+
+    protected abstract String getName();
 
     @Override
-    public JSONObject toJSON(){
+    public JSONObject toJSON() {
         JSONObject jsonObject = new JSONObject();
         //TODO
```
Comment (optionset) [inline app/src/main/java/org/mobilitychoices/entities/ListGenerator.java]: todo?

**Pull request merged**